test/profiling/MemoryTestPublisher.h

```diff
@@ -21,98 +21,159 @@
+    eprosima::fastdds::dds::Subscriber* subscriber_ {nullptr};
+    eprosima::fastdds::dds::TypeSupport command_type_ {new TestCommandDataType()};
+    eprosima::fastdds::dds::TypeSupport data_type_;
+    eprosima::fastdds::dds::Topic* command_pub_topic_ {nullptr};
+    eprosima::fastdds::dds::Topic* command_sub_topic_ {nullptr};
+    eprosima::fastdds::dds::Topic* data_topic_ {nullptr};
+    eprosima::fastdds::dds::DataWriter* command_writer_ {nullptr};
+    eprosima::fastdds::dds::DataWriter* data_writer_ {nullptr};
+    eprosima::fastdds::dds::DataReader* command_reader_ {nullptr};
+    int n_subscribers {0};
+    unsigned int n_samples {0};
     std::mutex mutex_;
-    int disc_count_;
+    int disc_count_ {0};
     std::condition_variable disc_cond_;
-    int comm_count_;
+    int comm_count_ {0};
     std::condition_variable comm_cond_;
-    int data_count_;
+    int data_count_ {0};
     std::condition_variable data_cond_;
-    int m_status;
-    unsigned int n_received;
-    bool n_export_csv;
+    int m_status {0};
+    unsigned int n_received {0};
+    bool n_export_csv {false};
     std::string m_exportPrefix;
-    bool init(int n_sub, int n_sam, bool reliable, uint32_t pid, bool hostname, bool export_csv,
-        const std::string& export_prefix,
-        const eprosima::fastrtps::rtps::PropertyPolicy& part_property_policy,
-        const eprosima::fastrtps::rtps::PropertyPolicy& property_policy,
-        const std::string& sXMLConfigFile, uint32_t data_size, bool dynamic_types);
-    void run(uint32_t test_time);
-    bool test(uint32_t test_time, uint32_t datasize);
-
-    class DataPubListener : public eprosima::fastrtps::PublisherListener
+
+    bool init(
+            int n_sub,
+            int n_sam,
+            bool reliable,
+            uint32_t pid,
+            bool hostname,
+            bool export_csv,
+            const std::string& export_prefix,
+            const eprosima::fastrtps::rtps::PropertyPolicy& part_property_policy,
+            const eprosima::fastrtps::rtps::PropertyPolicy& property_policy,
+            const std::string& sXMLConfigFile,
+            uint32_t data_size,
+            bool dynamic_types);
+    void run(
+            uint32_t test_time);
+    bool test(
+            uint32_t test_time,
+            uint32_t datasize);
+
+    class DataPubListener : public eprosima::fastdds::dds::DataWriterListener
     {
     public:
-        DataPubListener(MemoryTestPublisher* up) :mp_up(up), n_matched(0) {}
-        ~DataPubListener() {}
-        void onPublicationMatched(eprosima::fastrtps::Publisher* pub,
-            eprosima::fastrtps::rtps::MatchingInfo& info);
-        MemoryTestPublisher* mp_up;
-        int n_matched;
-    } m_datapublistener;
-
-    class CommandPubListener : public eprosima::fastrtps::PublisherListener
+
+        DataPubListener(
+                MemoryTestPublisher* up)
+            : up_(up)
+        {
+        }
+
+        ~DataPubListener()
+        {
+        }
```

Comment:
Please, apply this suggestion to any other empty destructor

```suggestion
        ~DataPubListener() = default;
```

Reply:
Done in d5af16ae0